atlassian/errors.go

```diff
@@ -0,0 +1,76 @@
+package atlassian
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/interpreter"
+)
+
+func (*Connector) interpretJSONError(res *http.Response, body []byte) error { //nolint:cyclop
+	payload := make(map[string]any)
+	if err := json.Unmarshal(body, &payload); err != nil {
+		return fmt.Errorf("interpretJSONError general: %w %w", interpreter.ErrUnmarshal, err)
+	}
+
+	// now we can choose which error response Schema we expect
+	var schema common.ErrorDescriptor
+
+	if _, ok := payload["status"]; ok {
+		apiError := &ResponseStatusError{}
+		if err := json.Unmarshal(body, &apiError); err != nil {
+			return fmt.Errorf("interpretJSONError SingleError: %w %w", interpreter.ErrUnmarshal, err)
+		}
+
+		schema = apiError
+	} else {
+		apiError := &ResponseMessagesError{}
```

Comment:
Good point!
There are multiple connectors that follow this strategy so all of them will have this bug.
![image](https://github.com/user-attachments/assets/7b5a1fea-c07b-499f-a80b-78de4340e0db)

Since this is not an immediate bug can we put this on my todo list? I will create a PR where I will take time to think how to handle error parsing and fallback in case underlying format changes.

Reply:
I created a dedicated PR for this issue. https://github.com/amp-labs/connectors/pull/840